### DISCUSSION TITLE
Local font : module not found on Vercel during build

### DISCUSSION THREAD

**Discussion by Alarid** [vercel/community#2367] [Help] [answered]
Following [the docs](https://nextjs.org/docs/app/building-your-application/optimizing/fonts#local-fonts) on local fonts, I added this in my `src/app/layout.tsx` : 

```ts
const adamBecker = localFont({
  src: [
    {
      path: './fonts/AdamBecker.ttf',
      weight: '400',
      style: 'normal',
    },
    {
      path: './fonts/AdamBecker Bold.ttf',
      weight: '700',
      style: 'bold',
    },
    {
      path: './fonts/AdamBecker-ExtraLight.ttf',
      weight: '200',
      style: 'extra-light',
    },
  ],
  variable: '--font-adam-becker',
})

const intro = localFont({
  src: './fonts/Intro.otf',
  variable: '--font-intro',
})

const bellania = localFont({
  src: './fonts/bellania.ttf',
  variable: '--font-bellania',
})
```

My fonts are located in the folder `src/app/fonts`. **Locally**, everything works as expected, even if I build the app. However, when I deploy the app on Vercel (after choosing the repo from Github and leaving the configuration as default) the build fails :

```
- info Creating an optimized production build...
--
21:24:25.235 | Failed to compile.
21:24:25.236 |  
21:24:25.236 | src/app/layout.tsx
21:24:25.236 | Module not found: Can't resolve './fonts/bellania.ttf'
21:24:25.236 |  
21:24:25.237 | https://nextjs.org/docs/messages/module-not-found
21:24:25.237 |  
21:24:25.237 | Import trace for requested module:
21:24:25.237 | ./src/app/layout.tsx
```

I tried removing the `./` from the `src` values, same error. Not sure why the path can't be resolve when building on Vercel ? How can I fix the build ?

**Comment by amyegan** [comment]
Hi @Alarid! I wasn't able to replicate this problem in my test app. Are you able to share a public repo or minimal reproduction?

**Reply by Alarid** [reply]
Hi @amyegan, I found out it wasn't a problem with Vercel, but with one of the fonts I was using. I'm not sure why, but when I tried a different font file (for the same font), the issue went away 🤷‍♂️

**Reply by amyegan** [reply]
I'm glad you found a way to get it working with a different font file. Not sure why the other one was failing, but at least you have a workaround!

**Reply by RamyAlshurafa** [reply]
@amyegan  it worked for me too,
.otf fails, but .ttf works

**Comment by jwcheon** [comment]
Having the same problem right now, builds well locally but on Vercel deploy runs into the same error message for some reason. Wonder why this happens... will try changing the font file :(

**Comment by mikemilla** [comment]
Still getting issues with this here. Any fix yet?

**Reply by jwcheon** [reply]
Hi @mikemilla I ended up downloading the same fonts but from a different source (one particular weight seemed to be an issue) + tried not changing the font file names, kinda annoying but that solved this issue for me to deploy into Vercel. Not sure what was the main reason was though still..

**Comment by decapination-labs** [comment]
i am having the same problem using MYRIADPRO-REGULAR.OTF.. 

Module not found: Can't resolve '../../public/fonts/MYRIADPRO-REGULAR.otf'

**Reply by christiantochen** [reply]
i had the same problem using `BAHNSCHRIFT.TTF`, 
and actually fixed it by changing filename and extensions into lowercase

**Reply by DevYemi** [reply]
@christiantochen Thank you, you suggestion worked for me

**Comment by ryancircelli** [comment]
I just solved this issue for myself!

The failing code:
`
const eight_track_hollow = localFont({
  src: "../../public/eight_hollow.ttf",
  variable: "--font-eight_track_hollow-font",
});
`

My file was named eight_hollow._**TTF**_, I renamed it to eight_hollow._**tff**_ and it was fixed. Seems to be a mismatch in casing between the file and import.

**Comment by smohammadhn** [comment, marked as answer]
The general rule of thumb here would be to avoid `spaces` and `uppercase` characters in both directory names and filenames.
so instead of `./fonts/Woff/Font Something.woff` => `./fonts/woff/font-something.woff`

This solved my issue, hope it works for you too.
